### PR TITLE
Replaced inline 3rd-party snippets with GTM container

### PR DESCRIPTION
**Pull request by lawrencejohnson**
The snippets that were baked into the `head.html` partial have been moved to a GTM container for developers.mattermost.com. That GTM container snippet has been added to the `head.html` and the `<noscript>` fallback has been added to all html layouts.

#### Ticket Link
https://mattermost.atlassian.net/browse/MAR-225

### CODE REVIEW

File: site/layouts/partials/head.html [hunk 1]
```diff
@@ -49,49 +49,11 @@
...
-<!-- Global site tag (gtag.js) - Google Analytics -->
-<script async src="https://www.googletagmanager.com/gtag/js?id=UA-64458817-2"></script>
-<script>
-    window.dataLayer = window.dataLayer || [];
-    function gtag() {dataLayer.push(arguments);}
-    gtag('js', new Date());
-    gtag('config', 'UA-64458817-2');
-</script>
 
-<!-- Marketo Munchkin Tracking code for Marketing Team -->
```
Comment (hanzei) [inline site/layouts/partials/head.html:62]: To double check: Are you sure these aren't used any longer?

File: site/layouts/partials/head.html [hunk 2]
```diff
@@ -49,49 +49,11 @@
 <link rel="stylesheet" href="{{ "css/note.css" | absURL }}">
 
 <script async defer src="https://buttons.github.io/buttons.js"></script>
-<script async src="//cdn.bizible.com/scripts/bizible.js"></script>
```
Comment (hanzei) [inline site/layouts/partials/head.html:52]: Same here

**Review by hanzei — CHANGES_REQUESTED** [review]
Two question to ensure we are not deleting anything that is still needed

File: site/layouts/partials/head.html (hunk 1 above)
Comment (lawrencejohnson) [reply site/layouts/partials/head.html:62]: Yes, they are still in use. I have moved them to the GTM container. If you inspect network connections when you load the page you can see the various calls are still there.

File: site/layouts/partials/head.html (hunk 2 above)
Comment (lawrencejohnson) [reply site/layouts/partials/head.html:52]: Same applies. Let me know if you have any questions. I can probably get you access to the GTM account if you'd like to review it there.

**Review by hanzei — APPROVED** [review]
LGTM :+1:

File: site/layouts/partials/head.html (hunk 2 above)
Comment (hanzei) [reply site/layouts/partials/head.html:52]: Got it. As long as they are in GTM, I'M down.

**Comment by lawrencejohnson** [comment]
Checking in on this PR.

**Comment by hanzei** [comment]
@lawrencejohnson Thanks for the ping. @aaronrothschild @justinegeffen Could one of you take a look?

**Review by aaronrothschild — APPROVED** [review]
lgtm

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
Thanks, @lawrencejohnson! LGTM!

**Pull request merged**